### PR TITLE
Fix JS checks

### PR DESCRIPTION
**Pull request by le0pard**

### CODE REVIEW

**Comment by bogdan** [comment]
We need a test.

File: lib/routes.js.coffee [hunk 1]
```diff
@@ -275,30 +275,27 @@ Utils =
...
 
   route_url: (route_defaults) ->
-    return route_defaults if typeof route_defaults == 'string'
+    return route_defaults if typeof route_defaults is 'string'
     protocol = route_defaults.protocol || Utils.current_protocol()
-    hostname = route_defaults.host || window.location.hostname
+    hostname = route_defaults.host || Utils.current_host()
     port = route_defaults.port || (Utils.current_port() unless route_defaults.host)
     port = if port then ":#{port}" else ''
+    "#{protocol}://#{hostname}#{port}"
```
Comment (krzysiek1507) [inline lib/routes.js.coffee]: it can produce:
`https://null:3000`

Comment (le0pard) [reply lib/routes.js.coffee]: @krzysiek1507 so if no hosts - no `"#{protocol}://#{hostname}#{port}"` at all? Or can be better solution?

Comment (krzysiek1507) [reply lib/routes.js.coffee]: @le0pard I was thinking about it and it seems that `hostname` has to be required for `_url` helpers in a `windowless` environment. It doesn't make sense to render HTML with URL like `https://null/products/bag` or `https:///products/bag`. Of course, it makes sense to render path but we want URL. Maybe we should follow Rails' implementation? What do you think?

**Comment by krzysiek1507** [comment]
Hi @le0pard I checked it again and I think we should fall back to path when `hostname` is missing

**Comment by krzysiek1507** [comment]
I've created PR #226

**Comment by le0pard** [comment]
@krzysiek1507 looks good. Closing this one

**Pull request closed without merging**